hack/create-identity-secret.sh

```diff
@@ -28,8 +28,17 @@ make --directory="${REPO_ROOT}" "${KUBECTL##*/}"
 # shellcheck source=hack/parse-prow-creds.sh
 source "${REPO_ROOT}/hack/parse-prow-creds.sh"
 
-if ! "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null; then
+if ! "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null && [[ -v "AZURE_CLIENT_SECRET" ]]; then
   "${KUBECTL}" create secret generic "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" --from-literal=clientSecret="${AZURE_CLIENT_SECRET}"
 fi
 
-"${KUBECTL}" label secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" "clusterctl.cluster.x-k8s.io/move-hierarchy=true" --overwrite=true
+if "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null; then
+  "${KUBECTL}" label secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" "clusterctl.cluster.x-k8s.io/move-hierarchy=true" --overwrite=true
+fi
+
+if ! "${KUBECTL}" get secret "${ASO_CREDENTIAL_SECRET_NAME}" > /dev/null; then
+  "${KUBECTL}" create secret generic "${ASO_CREDENTIAL_SECRET_NAME}" --from-literal=AZURE_SUBSCRIPTION_ID="${AZURE_SUBSCRIPTION_ID}" --from-literal=AZURE_TENANT_ID="${AZURE_TENANT_ID}" --from-literal=AZURE_CLIENT_ID="${AZURE_CLIENT_ID}" --from-literal=AUTH_MODE="${ASO_CREDENTIAL_SECRET_MODE:-workloadidentity}"
+fi
+
```

Comment:
should we add an `until` here as well before the label operation to account for the flows where we just created the secret a few ticks ago?

Reply:
I think the API is generally robust enough to correctly handle sequences like this with its own synchronization. Or have you seen a flake like that before?